### PR TITLE
Fix embedding loading, and remove defaults in model `from_params` methods

### PR DESCRIPTION
**Pull request by matt-gardner**
This PR _should_ get rid of crashes where loading a model would try to read a glove file that might not be there.  I'm not thrilled with how I had to create and pass around the `loading_saved_model` flag, but I'm not sure how else to address this.

While I was at it I also addressed #139, getting rid of the duplicate parameter specifications inside the code.  I'm not _totally_ certain this is a good idea, but I think it's better this way - we should just have one place to specify defaults, and I think the experiment config makes more sense than inside the code like it was, especially with issues around pre-trained embeddings files.

### CODE REVIEW

**Comment by joelgrus** [comment]
one alternative to `loading_saved_model` flag would be to manually remove the `"pretrained_file"` entries from the model config inside `Model.load()` before calling `Model.from_params()`. (Or alternatively to add a `"skip_loading": True` to the `embedding` part of the config and then check for that in `Embedding.from_config`.

those are both kind of hacky, not sure if they're preferable to passing a flag all the way down

**Comment by matt-gardner** [comment]
I very much think we should crash, not just give a warning, on initial model loading if we can't find an embedding file that you're wanting to use.  That's a fatal error.  Especially given how much info we spew out at the beginning of training, you'd almost certainly miss the warning, anyway.  And we _don't_ want to rely on the embedding file being there if you're loading a model - the weights are already saved in your weights file, and that introduces a really crazy runtime dependency, so in order to use our trained models you first have to download glove.  So we _have_ to have some kind of switch so that the model specification behaves slightly differently at train time vs. at load time.

I kind of like Joel's suggestion of just letting `Model.load()` handle this complexity, so that we don't have to muddy the rest of the API to handle this case.  One issue is that I'm thinking of adding some parameters that are only valid if you're using pre-trained embeddings, and you kind of want to remove all of those at the same time, which makes this a bit complicated...  But it's probably better than what I did here.

**Comment by matt-gardner** [comment]
This is ready for another look.

File: allennlp/models/model.py [hunk 1]
```diff
@@ -144,7 +144,10 @@ def load(cls,
         vocab_dir = os.path.join(serialization_prefix, 'vocabulary')
         vocab = Vocabulary.from_files(vocab_dir)
 
-        model = Model.from_params(vocab, config.get('model'))
+        model_params = config.get('model')
+
+        _remove_pretrained_embedding_params(model_params)
```
Comment (joelgrus) [inline allennlp/models/model.py]: I would just add a comment about why we had to do this

**Pull request merged**